src/bespokelabs/curator/request_processor/base_request_processor.py

```diff
@@ -262,6 +262,58 @@ def attempt_loading_cached_dataset(
                     "Deleted file and attempting to regenerate dataset from cached LLM responses."
                 )
 
+    def convert_response_to_response_format(
+        self, response_message: str | dict, response_format: Optional[BaseModel]
+    ) -> Optional[dict | str]:
+        """
+        Converts a response message to a specified Pydantic model format.
```

Comment:
A bit lost on the motivation here.

Reply:
this function we use anyways, I just moved it out so we don't duplicate code

We need to convert to the correct response format from the string that model returns

Reply:
Ah sorry.
Also, does it make sense to put this in prompt_formatter?
So `self.prompt_formatter.convert_response_to_response_format(response_message)`.